packages/react/src/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,33 +1,292 @@
+  const [effectiveHideRoot, setEffectiveHideRoot] = useState<boolean>(hideRoot)
+  //let effectiveOverflow = 'wrap'
+  const childArray = useMemo(() => getValidChildren(children), [children])
+
+  const rootItem = childArray[0]
+
+  const [visibleItems, setVisibleItems] = useState<React.ReactElement[]>(() => childArray)
+  const [childArrayWidths, setChildArrayWidths] = useState<number[]>(() => [])
+
+  const [menuItems, setMenuItems] = useState<React.ReactElement[]>([])
+  const [rootItemWidth, setRootItemWidth] = useState<number>(0)
+
+  // Menu button width with fallback
+  const MENU_BUTTON_FALLBACK_WIDTH = 32 // Design system small IconButton
+  const [menuButtonWidth, setMenuButtonWidth] = useState(MENU_BUTTON_FALLBACK_WIDTH)
+
+  // if (typeof window !== 'undefined') {
+  //   effectiveOverflow = overflow
+  // }
+  const MIN_VISIBLE_ITEMS = !effectiveHideRoot ? 3 : 4
+
+  useEffect(() => {
+    const listElement = containerRef.current?.querySelector('ol')
+    if (listElement && listElement.children.length > 0) {
+      const listElementArray = Array.from(listElement.children) as HTMLElement[]
+      const widths = listElementArray.map(child => child.offsetWidth)
+      setChildArrayWidths(widths)
+      setRootItemWidth(listElementArray[0].offsetWidth)
+    }
+  }, [childArray.length])
+
+  // Measure actual menu button width when it exists
+  useEffect(() => {
+    if (menuButtonRef.current) {
+      const measuredWidth = menuButtonRef.current.offsetWidth
+      if (measuredWidth > 0) {
+        setMenuButtonWidth(measuredWidth)
+      }
+    }
+  }, [menuItems.length]) // Re-measure when menu button appears/disappears
+
+  const calculateOverflow = useCallback(
+    (availableWidth: number) => {
+      const MENU_BUTTON_WIDTH = menuButtonWidth // Use measured width with fallback
+
+      const calculateVisibleItemsWidth = (w: number[]) => {
+        const widths = w.reduce((sum, width) => sum + width + 16, 0)
+        return !effectiveHideRoot ? rootItemWidth + widths : widths
+      }
+
+      let currentVisibleItems = [...childArray]
+      let currentVisibleItemWidths = [...childArrayWidths]
+      let currentMenuItems: React.ReactElement[] = []
+      let currentMenuItemsWidths: number[] = []
+      let eHideRoot = effectiveHideRoot
+
+      if (availableWidth > 0 && currentVisibleItemWidths.length > 0) {
+        let visibleItemsWidthTotal = calculateVisibleItemsWidth(currentVisibleItemWidths)
+
+        // Add menu button width if we have hidden items
+        if (currentMenuItems.length > 0) {
+          visibleItemsWidthTotal += MENU_BUTTON_WIDTH
+        }
+        while (
+          overflow === 'menu' &&
+          (visibleItemsWidthTotal > availableWidth || currentVisibleItems.length > MIN_VISIBLE_ITEMS)
+        ) {
+          // Remove the last visible item
+          const itemToHide = currentVisibleItems.slice(0)[0]
+          const itemToHideWidth = currentVisibleItemWidths.slice(0)[0]
```

Comment:
What's the call to `.slice(0)[0]` for here? Is it to do a deep copy on the item in the array or or something else?

Reply:
Not really, I was slicing a larger part in a prev implementation and ended up with this staler code.